core/capabilities/remote/trigger_publisher.go

```diff
@@ -72,6 +73,10 @@ func (p *triggerPublisher) Start(ctx context.Context) error {
 }
 
 func (p *triggerPublisher) Receive(_ context.Context, msg *types.MessageBody) {
+	if msg == nil {
```

Comment:
The fix I'm talking about would be done in this block here: https://github.com/smartcontractkit/chainlink/blob/94aa60db8f3842fef41f9afc2eb1bf94049b564b/core/capabilities/remote/dispatcher.go#L172

Reply:
target/client.go and target/server.go are indeed also affected

Reply:
I'm getting this warning message in the IDE when trying to remove the pointer from the `Receive()` definitions and calls: `'Receive' passes a lock by the value: type 'types.MessageBody' contains 'protoimpl.MessageState' contains 'sync.Mutex' which is 'sync.Locker'`.

Just sharing in case is important to note. It suggests using a pointer instead of just passing the value due to the lock copy that happens.

Reply:
Ah I overlooked that it was a proto -- in that case I'd just put the nil check in the caller